.github/workflows/production-migration.yml

```
@@ -0,0 +1,198 @@
+      - name: Require Mason's protected one-account approval boundary
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: node scripts/assert-production-environment-protection.mjs
+
+      - name: Bind dispatch to the current main commit
+        env:
+          EXPECTED_COMMIT: ${{ inputs.expected_commit }}
+          MIGRATION_NAME: ${{ inputs.migration_name }}
+          QUERY_SHA256: ${{ inputs.query_sha256 }}
+          REVIEWED_COMMIT: ${{ inputs.reviewed_commit }}
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2251:info:5:1: This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked

(shellcheck)